provisioning/baremetal_config.go

```diff
@@ -110,3 +117,80 @@ func validateDisabledConfig(prov *metal3iov1alpha1.Provisioning) error {
 	}
 	return nil
 }
+
+func getProvisioningIPCIDR(config *metal3iov1alpha1.ProvisioningSpec) *string {
+	if config.ProvisioningNetworkCIDR != "" && config.ProvisioningIP != "" {
+		_, net, err := net.ParseCIDR(config.ProvisioningNetworkCIDR)
+		if err == nil {
+			cidr, _ := net.Mask.Size()
+			ipCIDR := fmt.Sprintf("%s/%d", config.ProvisioningIP, cidr)
+			return &ipCIDR
+		}
+	}
+	return nil
+}
+
+func getDeployKernelUrl(config *metal3iov1alpha1.ProvisioningSpec) *string {
+	if config.ProvisioningIP != "" {
+		deployKernelUrl := fmt.Sprintf("http://%s/%s", net.JoinHostPort(config.ProvisioningIP, baremetalHttpPort), baremetalKernelUrlSubPath)
+		return &deployKernelUrl
+	}
+	return nil
+}
+
+func getDeployRamdiskUrl(config *metal3iov1alpha1.ProvisioningSpec) *string {
+	if config.ProvisioningIP != "" {
+		deployRamdiskUrl := fmt.Sprintf("http://%s/%s", net.JoinHostPort(config.ProvisioningIP, baremetalHttpPort), baremetalRamdiskUrlSubPath)
+		return &deployRamdiskUrl
+	}
+	return nil
+}
+
+func getIronicEndpoint(config *metal3iov1alpha1.ProvisioningSpec) *string {
+	if config.ProvisioningIP != "" {
+		ironicEndpoint := fmt.Sprintf("http://%s/%s", net.JoinHostPort(config.ProvisioningIP, baremetalIronicPort), baremetalIronicEndpointSubpath)
+		return &ironicEndpoint
+	}
+	return nil
+}
+
+func getIronicInspectorEndpoint(config *metal3iov1alpha1.ProvisioningSpec) *string {
+	if config.ProvisioningIP != "" {
+		inspectorEndpoint := fmt.Sprintf("http://%s/%s", net.JoinHostPort(config.ProvisioningIP, baremetalIronicInspectorPort), baremetalIronicEndpointSubpath)
+		return &inspectorEndpoint
+	}
+	return nil
+}
+
+func getProvisioningOSDownloadURL(config *metal3iov1alpha1.ProvisioningSpec) *string {
+	if config.ProvisioningOSDownloadURL != "" {
+		return &(config.ProvisioningOSDownloadURL)
+	}
+	return nil
+}
+
+func getMetal3DeploymentConfig(name string, baremetalConfig *metal3iov1alpha1.ProvisioningSpec) *string {
+	configValue := ""
+	switch name {
+	case "PROVISIONING_IP":
+		return getProvisioningIPCIDR(baremetalConfig)
+	case "PROVISIONING_INTERFACE":
+		return &baremetalConfig.ProvisioningInterface
+	case "DEPLOY_KERNEL_URL":
+		return getDeployKernelUrl(baremetalConfig)
+	case "DEPLOY_RAMDISK_URL":
+		return getDeployRamdiskUrl(baremetalConfig)
+	case "IRONIC_ENDPOINT":
+		return getIronicEndpoint(baremetalConfig)
+	case "IRONIC_INSPECTOR_ENDPOINT":
+		return getIronicInspectorEndpoint(baremetalConfig)
+	case "HTTP_PORT":
```

Comment:
import 	"k8s.io/utils/pointer"
return pointer.StringPtr(baremetalHttpPort)

Reply:
Is importing a new library to manage pointers giving me anything other than avoiding the use of a temp variable?

Reply:
Done.